apps/keyboard/js/views/layout_page_view.js

```diff
@@ -302,6 +302,25 @@ LayoutPageView.prototype.getNumberOfCandidatesPerRow = function() {
   return this.candidatePanel.countPerRow;
 };
 
+LayoutPageView.prototype.getHeight = function() {
+  var totalWidth = this.options.totalWidth;
+  var scale = this.viewManager.screenInPortraitMode() ?
+              totalWidth / 32 :
+              totalWidth / 64;
+
+  var height = this.rows.size * (5.1 * scale);
+
+  if (this.candidatePanel) {
+    if (this.candidatePanel instanceof LatinCandidatePanelView) {
```

Comment:
This is nice but it would be hard to set up in unit test.

How about moving `3.1` and `3.2` to a property of `candidatePanel` as constants?